### ISSUE TITLE
exception message = Undefined function 'TruncateTime'

### ISSUE THREAD

**Issue by walchandwarik** [bubibubi/JetEntityFrameworkProvider#5]
var pr = (from UserLog u in context.UserLogs
                          where u.UserId == UserId &&
                          DbFunctions.TruncateTime(u.LogInDate) == DateTime.Today.Date &&
                          u.LogOutDate == null
                          select u).ToList()

This query fails raising exception message "Undefined function 'TruncateTime' in expression."
But this query works well with MS Sql database

[Exception details.txt](https://github.com/bubibubi/JetEntityFrameworkProvider/files/1110211/Exception.details.txt)
[Inner Exception details.txt](https://github.com/bubibubi/JetEntityFrameworkProvider/files/1110210/Inner.Exception.details.txt)

**Comment by walchandwarik** [comment]
Dear sir,
Temporarily solved using following workaround
var pr = (from UserLog u in context.UserLogs
                          where u.UserId == UserId &&
                                u.LogInDate.Year == DateTime.Today.Year &&
                                u.LogInDate.Month == DateTime.Today.Month &&
                                u.LogInDate.Day == DateTime.Today.Day &&
                          u.LogOutDate == null
                          select u).ToList();

But above issue of "DbFunctions.TruncateTime" must be fixed.

Regards

**Comment by walchandwarik** [comment]
Dear sir, 
Even after new release "Added TruncateTime canonical function", the it throws the same exception of "Undefined function 'TruncateTime'"

[Inner Exception.txt](https://github.com/bubibubi/JetEntityFrameworkProvider/files/1112394/Inner.Exception.txt)
[Exception.txt](https://github.com/bubibubi/JetEntityFrameworkProvider/files/1112393/Exception.txt)

**Comment by bubibubi** [comment]
Did you compile from source code or did you download it with NuGet?
About NuGet, the provider will be fixed in next NuGet release (1.2.9).

Also, with TruncateTime, the generated SQL call a functions so if there is an index on the field it won't be used so I suggest you use a code similar to this:

```
var pr = (from UserLog u in context.UserLogs
 where u.UserId == UserId &&
 u.LogInDate >= DateTime.Today.Date &&  u.LogInDate < DateTime.Today.Date.AddDays(1) &&
 u.LogOutDate == null
 select u).ToList()
```

This code will use an index if you have one on LogInDate

**Comment by walchandwarik** [comment]
"Did you compile from source code or did you download it with NuGet?
About NuGet, the provider will be fixed in next NuGet release (1.2.9)."

Sorry I used from NuGet and release 1.2.8 which is latest till now. 
Waiting for 1.2.9.........
Thanks for your support

Regards,
